entities/vectorindex/hnsw/config.go

```diff
@@ -226,19 +240,25 @@ func optionalIntFromMap(in map[string]interface{}, name string,
 	// depending on whether we get the results from disk or from the REST API,
 	// numbers may be represented slightly differently
 	switch typed := value.(type) {
+	case int:
+		setFn(typed)
+		return nil
 	case json.Number:
 		asInt64, err = typed.Int64()
+		if err != nil {
```

Comment:
why this error check is now inside the switch? it was previously outside this part of code, bc it reacted on `errors` in the actual `switch` case, now it only reacts to errors in `json.Number` case...

Also sometimes switch returns value, sometimes not, sometimes parses the error and there's no default case also included, can this part be refactored in more structured way? so that it would be easy to read what it's being done and how we react to errors? now it's all mixed and hard to understand the flow

Reply:
1. The error checking is inside the switch statement now, because no other part of the switch statement can actually produce an error. 
2. I restructured it now, hopefully it is a more understandable.